packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -369,80 +375,44 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
-										sprintf( __( '%s Plan', __i18n_text_domain__ ), defaultPaidPlan?.title ?? '' )
-									}
-									badgeText={
-										defaultPaidPlan?.isPopular ? __( 'Popular', __i18n_text_domain__ ) : ''
-									}
-								/>
-								<TrailingContentSide nodeType="PRICE">
-									<span>{ defaultPaidPlan && planPrices[ defaultPaidPlan?.storeSlug ] }</span>
-									<span>
-										{
-											// translators: /mo is short for "per-month"
-											__( '/mo', __i18n_text_domain__ )
-										}
-									</span>
-								</TrailingContentSide>
-							</FocusedLaunchSummaryItem>
-
-							{ nonDefaultPaidPlan && (
-								<FocusedLaunchSummaryItem
-									onClick={ () => setPlan( nonDefaultPaidPlan ) }
-									isSelected={ isPlanSelected( nonDefaultPaidPlan ) }
-								>
-									<LeadingContentSide
-										label={
-											/* translators: %s is WordPress.com plan name (eg: Premium Plan) */
-											sprintf(
-												__( '%s Plan', __i18n_text_domain__ ),
-												nonDefaultPaidPlan.title ?? ''
-											)
-										}
-										badgeText={
-											nonDefaultPaidPlan.isPopular ? __( 'Popular', __i18n_text_domain__ ) : ''
-										}
-									/>
-									<TrailingContentSide nodeType="PRICE">
-										<span>{ planPrices[ nonDefaultPaidPlan?.storeSlug ] }</span>
-										<span>
-											{
-												// translators: /mo is short for "per-month"
-												__( '/mo', __i18n_text_domain__ )
-											}
-										</span>
-									</TrailingContentSide>
-								</FocusedLaunchSummaryItem>
+							{ allAvailablePlans.map(
+								( plan ) =>
+									plan && (
+										<FocusedLaunchSummaryItem
+											isLoading={ ! defaultFreePlan || ! defaultPaidPlan }
+											onClick={ () => plan && setPlan( plan ) }
+											isSelected={ plan && isPlanSelected( plan ) }
+											readOnly={ plan.isFree && ( hasPaidDomain || selectedPaidDomain ) }
+										>
+											<LeadingContentSide
+												label={
+													/* translators: %s is WordPress.com plan name (eg: Premium Plan) */
+													sprintf( __( '%s Plan', __i18n_text_domain__ ), plan?.title ?? '' )
+												}
+												badgeText={ plan?.isPopular ? __( 'Popular', __i18n_text_domain__ ) : '' }
+											/>
+											{ plan.isFree ? (
+												<TrailingContentSide
+													nodeType={ hasPaidDomain || selectedPaidDomain ? 'WARNING' : 'PRICE' }
+												>
+													{ hasPaidDomain || selectedPaidDomain
+														? __( 'Not available with your domain selection', __i18n_text_domain__ )
+														: __( 'Free', __i18n_text_domain__ ) }
+												</TrailingContentSide>
+											) : (
+												<TrailingContentSide nodeType="PRICE">
+													<span>{ plan && planPrices[ plan?.storeSlug ] }</span>
+													<span>
+														{
+															// translators: /mo is short for "per-month"
+															__( '/mo', __i18n_text_domain__ )
+														}
```

Comment:
I really think this is now less readable than before. 3-level conditionals that are pretty hard to wrap my head around.